src/files/XmlFile.ts

```diff
@@ -676,3 +464,67 @@ export class XmlFile {
         }
     }
 }
+
+function simpleMap(source: string, src: string) {
+    //create a source map from the original source code
+    let chunks = [] as (SourceNode | string)[];
+    let lines = src.split(/\r?\n/g);
+    for (let lineIndex = 0; lineIndex < lines.length; lineIndex++) {
+        let line = lines[lineIndex];
+        chunks.push(
+            lineIndex > 0 ? '\n' : '',
+            new SourceNode(lineIndex + 1, 0, source, line)
+        );
+    }
+
+    //sourcemap reference
+    chunks.push(`<!--//# sourceMappingURL=./${path.basename(source)}.map -->`);
+
+    return new SourceNode(null, null, source, chunks).toStringWithSourceMap();
+}
+
+function transpileAst(source: string, ast: SGAst, extraImports: string[]) {
```

Comment:
Why not call this `transpile` like the `BrsFile` equivalent? And have it operate on `this.ast` rather than having it passed in (again, like `BrsFile)